device_doctor/test/src/fake_android_device.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+
+import 'package:device_doctor/src/android_device.dart';
+
+class FakeAndroidDeviceDiscovery extends AndroidDeviceDiscovery {
```

Comment:
Shouldn't the tests be using the real implementation, but injecting the process manager to AndroidDevice?

Reply:
This is testing the real implementation by faking the very low level function `deviceListOutput` (migrated from earlier agent).
Injecting `process manager` will need a non-trivial change to existing logics. I would prefer to keep it as of now.

Reply:
Okay I see. In this case, we have AndroidDeviceDiscovery not able to be tested. I would be more comfortable with a refactor to allow testing the processes similar to how Flutter tools does it, but I defer to you to make that call.

Reply:
Injected, and have added a couple of more tests for utils.

Reply:
Thanks!